src/core/thread.d

```diff
@@ -631,7 +633,8 @@ class Thread
         }
 
         // Start thread as non-suspendable, undone in thread_entryPoint
-        atomicStore(m_isInCriticalRegion, true);
+        synchronized (Thread.criticalRegionLock)
+            m_isInCriticalRegion = true;
```

Comment:
> IIRC the thread had to be on the global list so that DllMain could detect that the thread is actually created by the runtime in the DLL_THREAD_ATTACH callback. Maybe this can be solved differently.

And also because the main thread has to wait for all started threads before exiting.